### PR TITLE
Support deleteAllIndexes in PersistentCacheIndexManager

### PR DESCRIPTION
**Pull request by cherylEnkidu**

### CODE REVIEW

**Review by dconeybe — COMMENTED** [review]
LGTM; however, I'll wait for someone more familiar with this code to give the formal LGTM.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/PersistentCacheIndexManager.java [hunk 1]
```diff
@@ -52,4 +52,12 @@ public void enableIndexAutoCreation() {
   public void disableIndexAutoCreation() {
     client.disableIndexAutoCreation();
   }
+
+  /**
+   * Removes all persistent cache indexes. Please note this function will also deletes indexes
+   * generated by {@link FirebaseFirestore#setIndexConfiguration(String)}.
```
Comment (wu-hui) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/PersistentCacheIndexManager.java:58]: We are removing this method, so no need to mention here.

**Review by wu-hui — APPROVED** [review]
LGTM with minor nit.

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/IndexingTest.java
Comment (dconeybe) [inline firebase-firestore/src/androidTest/java/com/google/firebase/firestore/IndexingTest.java:1]: Is there a test that calls deleteAllIndexes() after terminate()? If not, could you add one?

**Review by dconeybe — CHANGES_REQUESTED** [review]

File: firebase-firestore/src/main/java/com/google/firebase/firestore/PersistentCacheIndexManager.java (hunk 1 above)
Comment (cherylEnkidu) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/PersistentCacheIndexManager.java:57]: however when the feature releases, the deprecated setIndexConfiguration is still available to use. I would like to add notes mentioning the function is deprecated. Please let me know if you feel strong I should remove this comment.

**Pull request merged**